pkg/platform/runtime/artifact/download.go

```diff
@@ -63,3 +85,23 @@ func NewDownloadsFromCamelBuild(buildStatus *headchef_models.V1BuildStatusRespon
 
 	return nil, errs.New("No download found in build response: %+v", buildStatus)
 }
+
+func NewDownloadsFromCamelBuildPlan(build bpModel.Build, artifacts map[strfmt.UUID]ArtifactBuildPlan) ([]ArtifactDownload, error) {
+	for id := range artifacts {
+		for _, a := range build.Artifacts {
+			if a.Status == string(bpModel.ArtifactSucceeded) && a.TargetID == id && a.URL != "" {
+				if strings.Contains(a.URL, InstallerTestsSubstr) {
+					continue
+				}
+				if strings.HasPrefix(a.URL, "s3://as-builds/noop/") {
+					continue
+				}
+				if strings.HasSuffix(a.URL, ".tar.gz") || strings.HasSuffix(a.URL, ".zip") {
+					logging.Debug("Found download for artifact %s: %s", a.TargetID, a.URL)
+					return []ArtifactDownload{{ArtifactID: strfmt.UUID(a.TargetID), UnsignedURI: a.URL, UnsignedLogURI: a.LogURL, Checksum: a.Checksum}}, nil
+				}
```

Comment:
Shouldn't be necessary anymore with buildplans. We should have a definitive "this is the stuff you want" that's not based on evaluating string values for partial matches. Again; if that's not the case then we need to update the buildplan. That is very much a problem it is meant to solve.

Reply:
In testing these changes I noticed that for camel we need to filter by mime type. I've also updated the alternative implementation to filter on mime type as well since we _should_ be doing that.